### ISSUE TITLE
Collection JSON serialization

### ISSUE THREAD

**Issue by mmontone** [laravel/framework#10579]
I think JSON serialization of Collections is not right, at least not as right as it could be.

In the current implementation, the JSON serialization of a collection uses toArray to serialize, instead of recursively calling json_encode on the Collection items. The problem with that is that it is not enough to implement a jsonSerialize in the collection members, as it does not get called (but I think it should). Instead, you have to implement a toArray method in the collection members.

Current implementation:

```
    /**
     * Convert the object into something JSON serializable.
     *
     * @return array
     */
    public function jsonSerialize()
    {
        return $this->toArray();
    }
    /**
     * Get the collection of items as JSON.
     *
     * @param  int  $options
     * @return string
     */
    public function toJson($options = 0)
    {
        return json_encode($this->toArray(), $options);
    }
```

I propose to avoid the toArray call for implementing this. Instead of that, call json_encode directly on the collection members, and put that in a JSON array, or something along those lines (I'm no Laravel expert).

**Comment by rkgrep** [comment]
:thumbsup:

**Comment by rkgrep** [comment]
@mmontone Fixed in 5.2. You may close this issue.
